samples/cpp/fitellipse.cpp

```diff
@@ -276,39 +276,34 @@ void processImage(int /*h*/, void*)
     {
         vector<Point2f> pts = points[i];
 
-        if (pts.size()<=5) {
+        //At least 5 points can fit an ellipse
+        if (pts.size()<5) {
             continue;
         }
         if (fitEllipseQ) {
             box = fitEllipse(pts);
-            if( MAX(box.size.width, box.size.height) > MIN(box.size.width, box.size.height)*30 ||
-               MAX(box.size.width, box.size.height) <= 0 ||
-               MIN(box.size.width, box.size.height) <= 0){continue;};
+            //size.height >= size.width awalys,only if the pts are on a line or at the same point,size.width=0
```

Comment:
> size.height >= size.width awalys

Is this behavior built into `fitEllipse`?

Reply:
```C++
if( box.size.width > box.size.height )
{
    float tmp;
    CV_SWAP( box.size.width, box.size.height, tmp );
    box.angle = (float)(fmod((90 + theta*180/CV_PI),180.0)) ;
}
```
Yes,in the function of `fitEllipseDirect`、`fitEllipseNoDirect`、`fitEllipseAMS` include above code.
[https://github.com/opencv/opencv/blob/4.x/modules/imgproc/src/shapedescr.cpp](url)